core/primitives/src/errors.rs

```diff
@@ -1,8 +1,53 @@
+use crate::serialize::u128_dec_format;
 use crate::types::{AccountId, Balance, Nonce};
 use borsh::{BorshDeserialize, BorshSerialize};
 use near_crypto::PublicKey;
+use serde::{Deserialize, Serialize};
 use std::fmt::Display;
 
+use near_rpc_error_macro::RpcError;
+use near_vm_errors::VMError;
+
+/// Error returned in the ExecutionOutcome in case of failure.
+#[derive(
+    BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq, Eq, Deserialize, Serialize, RpcError,
+)]
+#[rpc_error_variant = "TxExecutionError"]
+pub enum ExecutionError {
```

Comment:
I vote for renaming the structure itself to `TxExecutionError` since `ExecutionError` is ambiguous even in the context of nearcore alone ("Execution" is used in at least two completely different contexts: Wasm execution and transaction execution)

@nearmax What do you think?

Reply:
Sounds good.

Reply:
renamed to TxExecutionError